src/app/hooks/useSearchQueryNetworkParam.ts

```diff
@@ -0,0 +1,29 @@
+import { useSearchParams } from 'react-router-dom'
+import { Network } from '../../types/network'
+import { RouteUtils } from '../utils/route-utils'
+
+/**
+ * Get the network from the URL search params and return it if it is valid,
+ * otherwise fallback to mainnet
+ */
+export const useSearchQueryNetworkParam = (): {
+  network: Network
+  setNetwork: (network: Network) => void
+} => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  let networkQueryParam = searchParams.get('network')
+  if (!networkQueryParam || !RouteUtils.getEnabledNetworks().includes(networkQueryParam as any)) {
+    networkQueryParam = Network.mainnet
+  }
```

Comment:
Also?
```suggestion
  const networkQueryParam = searchParams.get('network') ?? Network.mainnet
  if (!RouteUtils.getEnabledNetworks().includes(networkQueryParam as any)) {
    throw new Error('Unexpected network')
  }
```

Reply:
Yes... but we have a dedicated error type to throw for invalid network, so I will use that.